src/java.base/share/classes/java/io/Reader.java

```diff
@@ -198,6 +199,19 @@ public int read(char[] cbuf, int off, int len) throws IOException {
                 return n;
             }
 
+            @Override
+            public String readAllAsString() throws IOException {
+                ensureOpen();
+                String result = cs.toString().substring(next);
```

Comment:
For sake of efficiency can we please *first* limit and copy *last*? If `cs` refers to a huge object and `next` is just few bytes before `length()`, it makes no sense to copy the huge memory block just to return one or two characters finally. 🤔 

`result = cs.subSequence(next, cs.length()).toString()`?

Reply:
Improved in [04d8996](https://github.com/openjdk/jdk/pull/24728/commits/04d899631eb87dc475c7daf08f0ae7af47bcdac1).